src/schema/server_settings.py

```diff
@@ -99,87 +99,287 @@ class Difficulty(str, Enum):
+    },
+    {
+        "value": "pixel_art_3",
+        "label": "Pixel Art 3 (Stable Diffusion)",
+        "imageSample": "/image_samples/pixel_art_3.png",
+    },
+    {
+        "value": "dall_e_2_standard",
+        "label": "DALL-E 2",
+        "imageSample": "/image_samples/dall_e_2_standard.png",
+    },
+    {
+        "value": "dall_e_2_stellar_dream",
+        "label": "Stellar Dream (DALL-E 2)",
+        "imageSample": "/image_samples/dall_e_2_stellar_dream.png",
+    },
+    {
+        "value": "dall_e_2_neon_cyberpunk",
+        "label": "Neon Cyberpunk (DALL-E 2)",
+        "imageSample": "/image_samples/dall_e_2_neon_cyberpunk.png",
+    },
+    {
+        "value": "cinematic_animation",
+        "label": "Outdoor Fantasy Painting (Stable Diffusion)",
+        "imageSample": "/image_samples/cinematic_animation.jpeg",
+    },
+]
+
+
 class ServerSettings(BaseModel):
     """Server Settings for the AI Adventure Game set by the Game Host.
 
     These are intended to be set by the game operator (not the user).
     """
 
+    @classmethod
+    def setting_schema(cls, name: str) -> Dict[str, any]:
+        field = cls.__fields__.get(name)
+        if not field:
+            raise AttributeError(f"No field named {name}")
+        meta_setting = field.field_info.extra.get("meta_setting")
+        if not meta_setting:
+            raise AttributeError(f"Field {name} has no setting schema")
+        # See TODO on name
+        # meta_setting["default"] = field.default
+        meta_setting["serverSetting"] = True
+        return meta_setting
+
     # Language Generation Settings - Function calling
     default_function_capable_llm_model: str = Field("gpt-3.5-turbo", description="")
     default_function_capable_llm_temperature: float = Field(0.0, description="")
     default_function_capable_llm_max_tokens: int = Field(512, description="")
 
     # Language Generation Settings - Story telling
-    default_story_model: str = Field("gpt-3.5-turbo", description="")
-    default_story_temperature: float = Field(0.7, description="")
-    default_story_max_tokens: int = Field(512, description="")
+    default_story_model: str = Field(
+        "gpt-3.5-turbo",
+        meta_setting={
+            # Validated
+            "name": "default_story_model",
+            "label": "Story LLM Model",
+            "description": "Model used to generate story text.",
+            "type": "select",
+            "default": "gpt-3.5-turbo",
+            "options": [
+                {
+                    "value": "gpt-3.5-turbo",
+                    "label": "GPT 3.5 Turbo",
+                },
+                {
+                    "value": "gpt-4",
+                    "label": "GPT 4",
+                },
+            ],
+        }
+    )
+    default_story_temperature: float = Field(
+        0.7,
+        meta_setting={
+            # NEEDS WORK:
+            # TODO: Add a post-processing step to coerce this to a float.
```

Comment:
That makes sense to me -- would this be on ingress from the web client? (e.g. if `type=float` then coerce to float?)

Reply:
FWIW this was existing commentary that came over IIRC from the web itself.  It might not be relevant anymore.